deployer/scripts/upgrade.sh

```diff
@@ -489,22 +489,37 @@ function add_fluentd_daemonset() {
 function add_config_maps() {
   generate_configmaps
   echo "Supplying Elasticsearch with a ConfigMap..."
+  patchPIDs=()
   local dc patch=$(join , \
     '{"op": "replace", "path": "/spec/template/spec/containers/0/volumeMounts/0/mountPath", "value": "/etc/elasticsearch/secret"}' \
     '{"op": "add", "path": "/spec/template/spec/containers/0/volumeMounts/1", "value": {"name": "elasticsearch-config", "mountPath": "/usr/share/elasticsearch/config", "readOnly": true}}' \
-    '{"op": "add", "path": "/spec/template/spec/volumes/1", "value": {"name": "elasticsearch-config", "configMap": {"name": "elasticsearch-config"}}}' \
+    '{"op": "add", "path": "/spec/template/spec/volumes/1", "value": {"name": "elasticsearch-config", "configMap": {"name": "logging-elasticsearch"}}}' \
   )
   for dc in $(get_es_dcs); do
+    currentVersion=$(oc get $dc -o jsonpath='{.status.latestVersion}')
     oc patch $dc --type=json --patch "[$patch]"
+
+    oc deploy $dc --latest
+    waitForChange $currentVersion $dc &
+    patchPIDs+=( $!)
   done
   echo "Supplying Curator with a ConfigMap..."
   patch=$(join , \
     '{"op": "add", "path": "/spec/template/spec/containers/0/volumeMounts/1", "value": {"name": "config", "mountPath": "/etc/curator/settings", "readOnly": true}}' \
-    '{"op": "add", "path": "/spec/template/spec/volumes/1", "value": {"name": "config", "configMap": {"name": "curator-config"}}}' \
+    '{"op": "add", "path": "/spec/template/spec/volumes/1", "value": {"name": "config", "configMap": {"name": "logging-curator"}}}' \
   )
   for dc in $(get_curator_dcs); do
-    oc patch $dc --type=json --patch "[$patch]"
+    # we want to ignore a failure here for the scenario where we are installing Curator as part of this upgrade
+    # since the curator will already contain the changes oc patch provides and will fail due to duplicate volume and mountPath
+    currentVersion=$(oc get $dc -o jsonpath='{.status.latestVersion}')
+    if oc patch $dc --type=json --patch "[$patch]"; then
+      oc deploy $dc --latest
+      waitForChange $currentVersion $dc &
+      patchPIDs+=( $!)
+    fi
   done
+
+  wait ${patchPIDs[@]}
```

Comment:
i believe `oc deploy` will fail in the case where there is currently a deployment scheduled for `--latest`.  In that case the output _should_ be captured in the deployer log file.  Maybe we should have `oc deploy $dc --latest && waitForChange ...` ?